recipes/libsass/all/conanfile.py

```diff
@@ -41,107 +39,134 @@ def config_options(self):
-        return self._autotools
-
-    def _build_autotools(self):
-        with chdir(self, self._source_subfolder):
-            save(self, path="VERSION", content=f"{self.version}")
-            self.run("{} -fiv".format(tools.get_env("AUTORECONF")))
-            autotools = self._configure_autotools()
-            autotools.make()
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     @property
-    def _make_program(self):
-        return tools.get_env("CONAN_MAKE_PROGRAM", tools.which("make") or tools.which("mingw32-make"))
-
-    def _build_mingw(self):
-        makefile = os.path.join(self._source_subfolder, "Makefile")
-        replace_in_file(self, makefile, "CFLAGS   += -O2", "")
-        replace_in_file(self, makefile, "CXXFLAGS += -O2", "")
-        replace_in_file(self, makefile, "LDFLAGS  += -O2", "")
-        with chdir(self, self._source_subfolder):
-            env_vars = AutoToolsBuildEnvironment(self).vars
-            env_vars.update({
-                "BUILD": "shared" if self.options.shared else "static",
-                "PREFIX": tools.unix_path(os.path.join(self.package_folder)),
+    def _msbuild_configuration(self):
+        return "Debug" if self.settings.build_type == "Debug" else "Release"
+
+    def generate(self):
+        if not is_msvc(self):
+            tc = AutotoolsToolchain(self)
+            tc.configure_args += ["--disable-tests"]
+            env = tc.environment()
+            if self._is_mingw:
+                env.define("BUILD", "shared" if self.options.shared else "static")
                 # Don't force static link to mingw libs, leave this decision to consumer (through LDFLAGS in env)
-                "STATIC_ALL": "0",
-                "STATIC_LIBGCC": "0",
-                "STATIC_LIBSTDCPP": "0",
-            })
-            with tools.environment_append(env_vars):
-                self.run(f"{self._make_program} -f Makefile")
-
-    def _build_visual_studio(self):
-        with chdir(self, self._source_subfolder):
-            properties = {
-                "LIBSASS_STATIC_LIB": "" if self.options.shared else "true",
-                "WholeProgramOptimization": "true" if any(re.finditer("(^| )[/-]GL($| )", tools.get_env("CFLAGS", ""))) else "false",
-            }
-            platforms = {
-                "x86": "Win32",
-                "x86_64": "Win64"
-            }
-            msbuild = MSBuild(self)
-            msbuild.build(os.path.join("win", "libsass.sln"), platforms=platforms, properties=properties)
+                env.define("STATIC_ALL", "0")
+                env.define("STATIC_LIBGCC", "0")
+                env.define("STATIC_LIBSTDCPP", "0")
+            tc.generate(env)
+        else:
+            with chdir(self, self.source_folder):
+                tc = MSBuildToolchain(self)
+                tc.configuration = self._msbuild_configuration
+                tc.platform = "Win32" if self.settings.arch == "x86" else "Win64"
```

Comment:
This attribute is no-op, there is no platform attribute in `MSBuildToolchain` class